boofuzz/primitives/string.py

```diff
@@ -221,26 +220,49 @@ def _yield_long_strings(self, sequences):
         @param sequences: Sequence to repeat for creation of fuzz strings.
         """
         for sequence in sequences:
-            for size in self._long_string_lengths:
-                for delta in self._long_string_deltas:
-                    yield sequence * (size + delta)
-
-        for size in self._long_string_lengths:
-            s = "D" * size
-            for loc in self.random_indices[size]:
-                s = s[:loc] + "\x00" + s[loc:]
-                yield s
+            for size in [
+                length + delta
+                for length, delta in itertools.product(self._long_string_lengths, self._long_string_deltas)
+            ]:
+                if self.max_len is None or size <= self.max_len:
+                    data = sequence * math.ceil(size / len(sequence))
+                    yield data[:size]
+                else:
+                    break
 
-        for sequence in sequences:
             for size in self._extra_long_string_lengths:
-                yield sequence * size
+                if self.max_len is None or size <= self.max_len:
+                    data = sequence * math.ceil(size / len(sequence))
+                    yield data[:size]
+                else:
+                    break
+
+            if self.max_len is not None:
+                data = sequence * math.ceil(self.max_len / len(sequence))
+                yield data
+
+        for size in self._long_string_lengths:
+            if self.max_len is None or size <= self.max_len + 1:
+                s = "D" * size
+                for loc in self.random_indices[size]:
+                    s = s[:loc] + "\x00" + s[loc:]  # Do we really have to insert a char or can we replace it?
```

Comment:
Do we really have to insert a char or can we replace it at the given location?

Reply:
Replacement should work! Probably a silly carry-over as far as I can tell.

Reply:
Perfect, that will make things a little easier.
I'll change it.